### PR TITLE
Adds metrics for tracking when instrumentation is applied 

### PR DESCRIPTION
**Pull request by michaelgoin**
## Proposed Release Notes

* Added support metrics for tracking when instrumentation was applied per module.

  * `Supportability/Features/Instrumentation/OnResolved/<module-name>`
  * `Supportability/Features/Instrumentation/OnResolved/<module-name>/Version/<major version>`
  * `Supportability/Features/Instrumentation/OnRequire/<module-name>`
  * `Supportability/Features/Instrumentation/OnRequire/<module-name>/Version/<major version>`

## Details

Examples:

- 'Supportability/Features/Instrumentation/OnRequire/domain', 
- 'Supportability/Features/Instrumentation/OnRequire/restify', 
- 'Supportability/Features/Instrumentation/OnRequire/restify/Version/6'

Only attempts to grab the version if the shim found a module root to avoid grabbing the agent version. Version doesn't apply to 'domain' either, for example.

### CODE REVIEW

File: lib/shimmer.js [hunk 1]
```diff
@@ -648,3 +654,35 @@ function hasValidRegisterOptions(opts) {
 
   return true
 }
+
+function trackInstrumentationUsage(agent, shim, moduleName, metricPrefix) {
```
Comment (bizob2828) [inline lib/shimmer.js:658]: mind adding some jsdoc here?  I also would love to see some unit tests for this code.  I do notice we have versioned tests that indirectly verify this but might be better for documentation and future selves

**Review by bizob2828 — CHANGES_REQUESTED** [review]
💯 this will def add some value for both us and support when digging into usage

File: lib/shimmer.js (hunk 1 above)
Comment (michaelgoin) [reply lib/shimmer.js:658]: Sure, I can add tests. We often don't for support metrics that aren't a part of a shared spec as these tend to be a bit more subject to change and removal but in this case we are looking to potentially leverage to drive decision making so makes sense to be a bit more careful.

jsdoc is fine as well although unlike to add much value.

File: test/integration/module-loading/module-loading.tap.js [hunk 2]
```diff
@@ -75,14 +79,118 @@ tap.test('shim.require() should play well with multiple test runs', (t) => {
   t.end()
 })
 
+tap.test('Should create usage metric onResolved', (t) => {
```
Comment (bizob2828) [inline test/integration/module-loading/module-loading.tap.js:82]: nice. i was just thinking to write unit tests but this is def better.

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**